docs/source/coding_guidelines.md

```diff
@@ -0,0 +1,253 @@
+called, for example, after an AzureML run to collect results, can this be automated further, and make the script
+obsolete? Reasons for this approach:
+
+* People tend to forget that there is a script to do X already, and may re-do the task in question manually.
+* Any script that requires input from the user also has a chance to be provided with the wrong input, leading to friction or incorrect results.
+  In a programmatic scenario, where the script is called automatically, this chance of errors is greatly minimized.
+
+## What not to commit
+
+* DO NOT commit files taken from or derived from private datasets.
+* Avoid checking in large files (anything over 100 kB).
+  If you need to commit large files, consider adding them via [Git LFS](https://git-lfs.github.com/).
+
+### Jupyter Notebooks
+
+Notebooks can easily become obsolete over time and may not work with code changes.
+Also, testing them is generally difficult.
+Please follow these guidelines for notebooks:
+
+* If you are planning to use a notebook, avoid committing into the repository unless it is part of a project demo.
+* If the notebook is used to document results, then please render the notebook and place the results in a separate
+  document outside the repository.
+* Bear in mind that notebooks are also an easy way of leaking sensitive data: They can for example contain images
+  derived from private datasets.
+  You should clear the notebook outputs before committing it.
+
+## Review / Pull Requests
+
+### Scope of Pull Requests
+
+PRs should ideally implement a single change.
+If in doubt, err on the side of making the PR too small, rather than too big.
+
+* Small PRs help reduce the load on the reviewer.
+* Avoid adding unrelated changes to an existing PR.
+* PRs should be modular: we can iterate on PRs, and any positive delta is a contribution.
+
+Please follow the guidance on
+[Github flow](https://docs.github.com/en/get-started/quickstart/github-flow).
+
+Try gauging the value of your contribution for yourself by asking the following questions:
+
+* Will this change bring the team closer to achieving their project goals?
+* Will someone else understand my code?
+* Will they be able to use my code?
+* Will they be able to extend or build on top of my code?
+
+### Pull Request Process
+
+* When creating a PR, do add a summary of your contribution in the PR description.
+* The template PR description also contains a checklist for the PR author.
+* For collecting early feedback on your work, please use a Draft Pull Request.
+  These PRs are marked with a grey icon in the Github UI, and send a clear signal that the code there is not yet ready for review.
+  When submitting a draft PR, all the checks will be run.
+* Once your work is ready for review, click the "Ready for Review" button on the Github PR page, and, if you want, assign reviewers for your PR.
+
+### Pull Request Contents
```

Comment:
I'm not sure this section is necessary - the only new point here I think is linking to an issue, which could be made part of the above section on Pull Request Process